### PR TITLE
Add some group plugin APIs

### PR DESCRIPTION
**Pull request by jwilander**
#### Summary
Add some group plugin APIs. Context: https://community.mattermost.com/core/pl/jumrifixityx7d74g88uwng3sr

### CODE REVIEW

File: plugin/api.go [hunk 1]
```diff
@@ -289,6 +289,15 @@ type API interface {
 	// UpdateChannelMemberNotifications updates a user's notification properties for a channel.
 	UpdateChannelMemberNotifications(channelId, userId string, notifications map[string]string) (*model.ChannelMember, *model.AppError)
 
+	// GetGroup gets a group by ID.
```
Comment (ali-farooq0) [inline plugin/api.go]: should these contain `Minimum server version: 5.16` comment?

Comment (ali-farooq0) [reply plugin/api.go]: Also what happens if these are invoked on team-edition and not EE-dition? Do they get an empty result back or an error that it's not implemented?

**Comment by prapti** [comment]
I'm going to wait reviewing till all devs approve.

File: plugin/api.go (hunk 1 above)
Comment (jwilander) [reply plugin/api.go]: Yes good catch, I'll add the comments.

If they are used by TE installation then they'll just return an error, similar to other plugin APIs that are for EE features like LDAP.

Comment (jwilander) [reply plugin/api.go]: Added for 5.18 as this missed the 5.16 deadline

**Comment by jwilander** [comment]
@prapti ready for you

**Comment by prapti** [comment]
Checking with Dylan Haussermann if he can help with testing this one.

**Comment by jwilander** [comment]
Any update on this @prapti ?

**Comment by prapti** [comment]
@jwilander Testing in progress.

**Pull request merged**